src/py/flwr/client/app.py

```diff
@@ -97,8 +97,18 @@ def run_client_app() -> None:
     if client_app_dir is not None:
         sys.path.insert(0, client_app_dir)
 
+    valid, error_msg = validate(getattr(args, "client-app"))
+    if not valid and error_msg:
+        raise LoadClientAppError(error_msg) from None
+
     def _load() -> ClientApp:
-        client_app: ClientApp = load_client_app(getattr(args, "client-app"))
+        client_app = load_app(getattr(args, "client-app"), LoadClientAppError)
+
+        if not isinstance(client_app, ClientApp):
+            raise LoadClientAppError(
+                f"Attribute {getattr(args, 'client-app')} is not of type {ClientApp}",
+            ) from None
```

Comment:
Given that we have this check after each of our three call to `load_app`, we can also move it into `load_app` by passing `ClientApp` or `ServerApp` as and additional argument.

Reply:
This created issues with circular imports (for type hints)